README.md

```diff
@@ -1,3 +1,231 @@
-#gateway-python-sdk
+# gateway-python-sdk
 
-Python SDK for DevIoT gateway service
+### Python SDK for DevIoT service ([https://deviot.cisco.com/](https://deviot.cisco.com/))
+
+You can use this SDK to register devices to DevIoT and sync up data and actions between the devices and DevIoT.
+
+## Table of contents
+
+* [Getting Started](#getting-started)
+* [API](#api)
+
+## Requirement
+1. [Python](https://www.python.org/downloads/): This SDK is written in Python. We support both Python 2 and 3.
+2. [paho-mqtt](https://eclipse.org/paho/clients/python/): This SDK uses this library to build a MQTT client
+
+## Usage
+1. You can use SDK to register your own sensors and systems to DevIoT.
+2. Use [Starter-ki](https://wwwin-github.cisco.com/DevIoT/gateway-python-starter-kit) to run sample codes
+
+## Term
+
+- **Gateway**: 'Gateway' is a device connected to DevIoT like a Raspberry Pi board or a mobile phone
+- **Thing**: 'Thing' is a sensor or a module in a gateway like a light sensor, LED, or GPS. A thing is an icon in DevIoT. A gateway can have several things.
+- **Property**: 'Property' is a variable measured by a thing. For instance, Latitude and longitude are properties of a GPS. A thing can have several properties.
+- **Action**: 'Action' is a command for a module. For example, turning on and off LED are two different actions. A thing can have several actions.
+
+## Getting Started
+#### 0) Check the version of python (terminal)
+Before using this SDK, you need to check the version of python. The master branch is written in **Python 2**. You can check it with the following command.
+```
+python --version
+```
+
+If your python version is 3.x, you need to switch to 'python3' branch in the next step.
+
+
+#### 1) Set up SDK package (terminal)
+Clone SDK git repository
+```
+git clone https://github.com/ailuropoda0/gateway-python-sdk.git
+cd gateway-python-sdk
+```
+(**ONLY for Python 3 users**) Switch to Python 3 
+```
+git fetch origin python3:python3
+git checkout python3
+```
+
+Install sdk package on python
+```
+python setup.py install
+```
+
+#### 2) Connect gateway (python code)
+You can refer the example code in [gateway-python-starter-kit](https://wwwin-github.cisco.com/DevIoT/gateway-python-starter-kit).
+
+Import SDK
+```
+from cisco_deviot.gateway import Gateway
+from cisco_deviot.thing import Thing, Property, PropertyType
+```
+Construct a Gateway object
+```
+account = "your_id@cisco.com" # enter your DevIoT account
+app = Gateway(name="gateway_name", account=account)
+```
+
+Contruct a thing instance
+```
+thing = Thing("thing-id", "thing-name", "thing-kind")
+```
+You can only use Thing as either an input component(sensor) or an output component.
+If your thing has both properties and actions, it acts as an input component.
```

Comment:
I think if these kinds of thing can be used as both input/output component

Reply:
I have checked a Thing instance can be both input and output component.
But I think I can only use a Thing instance as either input or output component in a project page.

So I rewrite this part like the below.

> A Thing instance can have both *properties* and *actions*. But you can only use it as *either* an input component *or* an output component in each project.